contracts/BullaInstaPay.sol

```diff
@@ -0,0 +1,81 @@
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/IBullaManager.sol";
+
+struct Multihash {
+    bytes32 hash;
+    uint8 hashFunction;
+    uint8 size;
+}
+
+error ZeroAddress();
+error ValueMustBeGreaterThanZero();
+error ClaimTokenNotContract();
+
+contract BullaInstantPayment {
+    using Address for address;
+    using SafeERC20 for IERC20;
+
+    address public bullaManager;
+
+    constructor(address _bullaManager) {
+        bullaManager = _bullaManager;
+    }
+
+    event InstantPayment (
+        address bullaManager,
+        address indexed from,
+        address indexed to,
+        uint256 amount,
+        address tokenAddress,
+        string description,
+        string[] tags,
+        uint256 blocktime,
+        Multihash attachment
+    );
+
+    function instantPayment(
+        address to,
+        uint256 amount,
+        address tokenAddress,
+        string memory description,
+        string[] memory tags,
+        Multihash calldata attachment
+    ) public {
+        if (to == address(0)) {
+            revert ZeroAddress();
+        }
+
+        if (amount == 0) {
+            revert ValueMustBeGreaterThanZero();
+        }
+
+        if (!tokenAddress.isContract()) {
```

Comment:
We can avoid this check too. if we call safeTransferFrom on the tokenAddress and that contract doesn't have that method, or reverts, this whole tx will revert. The SafeERC20 lib handles some of this nicely for us